process_scheduling_algorithms/non_preemptive_priority_scheduling.c

```diff
@@ -0,0 +1,419 @@
+#include <stdio.h>    /// for IO operations (`printf`)
+#include <stdlib.h>  /// for memory allocation eg: `malloc`, `realloc`, `free`, `exit`
+/**
+ * @brief Structure to represent a process
+ */
+struct node
+{
+    int ID;             ///< ID of the process node
+    int AT;             ///< Arrival Time of the process node
+    int BT;             ///< Burst Time of the process node
+    int priority;       ///< Priority of the process node
+    int CT;             ///< Completion Time of the process node
+    int WT;             ///< Waiting Time of the process node
+    int TAT;            ///< Turn Around Time of the process node
+    struct node *next;  ///< pointer to the node
+};
+
+/**
+ * @brief To insert a new process in the queue
+ * @param root pointer to the head of the queue
+ * @param id process ID
+ * @param at arrival time
+ * @param bt burst time
+ * @param prior priority of the process
+ * @returns void
+ */
+void insert(struct node **root, int id, int at, int bt, int prior)
+{
+    // create a new node and initialize it
+    struct node *new = (struct node *)malloc(sizeof(struct node));
+    struct node *ptr = *root;
+    new->ID = id;
+    new->AT = at;
+    new->BT = bt;
+    new->priority = prior;
+    new->next = NULL;
+    new->CT = 0;
+    new->WT = 0;
+    new->TAT = 0;
+    // if the root is null, make the new node the root
+    if (*root == NULL)
+    {
+        *root = new;
+        return;
+    }
+    // else traverse to the end of the queue and insert the new node there
+    while (ptr->next != NULL)
+    {
+        ptr = ptr->next;
+    }
+    ptr->next = new;
+    return;
+}
+/*
+ * @brief To delete a process from the queue
+ * @param root pointer to the head of the queue
+ * @param id process ID
+ * @returns void
+ */
+void delete(struct node **root, int id)
+{
+    struct node *ptr = *root, *prev;
+    // if the root is null, return
+    if (ptr == NULL)
+    {
+        return;
+    }
+    // if the root is the process to be deleted, make the next node the root
+    if (ptr->ID == id)
+    {
+        *root = ptr->next;
+        free(ptr);
+        return;
+    }
+    // else traverse the queue and delete the process
+    while (ptr != NULL && ptr->ID != id)
+    {
+        prev = ptr;
+        ptr = ptr->next;
+    }
+    if (ptr == NULL)
+    {
+        return;
+    }
+    prev->next = ptr->next;
+    free(ptr);
+}
+/**
+ * @brief To show the process queue
+ * @param head pointer to the head of the queue
+ * @returns void
+ */
+void showList(struct node *head)
+{
+    printf("Process Priority AT BT CT TAT WT \n");
+    while (head != NULL)
+    {
+        printf("P%d. %d %d %d %d %d %d \n", head->ID, head->priority, head->AT,
+               head->BT, head->CT, head->TAT, head->WT);
+        head = head->next;
+    }
+}
+/**
+ * @brief To length process queue
+ * @param root pointer to the head of the queue
+ * @returns int total length of the queue
+ */
+int LLength(struct node **root)
```

Comment:
In the contribution guidelines, it states that the function names must be `snake_cased` please revise your PR to meet that standard. (you also need to change the `CalculateCT` function.)